src/client/profile-provider.test.ts

```diff
@@ -47,6 +47,8 @@ jest.mock('fs', () => ({
   ...jest.requireActual<Record<string, unknown>>('fs'),
   promises: {
     readFile: jest.fn(),
+    mkdir: jest.fn(),
```

Comment:
No more tests?

Reply:
I'd spend an hour mocking everything and then just delete it in refactor when I'm rewriting it, so I don't think the value is there :shrug:

Reply:
Hmm, hour of mocking sucks but it is not cool to not have tests 😩
I am ok with writing them (Friday or Monday) but we need to break this file (during your refactor) into more testable pieces.

Reply:
I added tests. Hope you're happy.